### PR TITLE
EVG-13443: Implement mock sender for testing

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-13443

### CODE REVIEW

File: send/mock.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+// NewMockSender returns a MockSender with the given name.
+func NewMockSender(name string) *MockSender {
+	return &MockSender{
+		Base: NewBase(name),
+	}
+}
+
+func (s *MockSender) Send(m message.Composer) {
```
Comment (Kimchelly) [inline send/mock.go:28]: Documentation.

File: send/mock.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	s.Messages = append(s.Messages, m)
+}
+
+func (s *MockSender) Flush(_ context.Context) error {
```
Comment (Kimchelly) [inline send/mock.go:32]: Documentation.

File: send/mock.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	if s.FlushErr {
+		return errors.New("flush error")
+	}
+
+	return nil
+}
+
+func (s *MockSender) Close() error {
```
Comment (Kimchelly) [inline send/mock.go:40]: Documentation.

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: send/mock.go (hunk 1 above)
Comment (julianedwards) [reply send/mock.go:28]: I thought we don't document interfaces?

Comment (Kimchelly) [reply send/mock.go:28]: My rule is to document exported types since they're publicly visible.

Comment (julianedwards) [reply send/mock.go:28]: Done.

File: send/mock.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+// returned.
+func (s *MockSender) Flush(_ context.Context) error {
+	if s.FlushErr {
+		return errors.New("flush error")
+	}
+
+	return nil
+}
+
+// Close sets Closed to true. If either CloseErr or Close are set to true, an
```
Comment (Kimchelly) [inline send/mock.go:42]: or Closed*

**Review by Kimchelly — APPROVED** [review]
LGTM mod typo.

**Pull request merged**